sharktank/sharktank/layers/paged_llama_attention_block.py

```diff
@@ -147,16 +154,16 @@ def repeat_kv(x: torch.Tensor) -> torch.Tensor:
         keys = xk.transpose(1, 2)
         values = xv.transpose(1, 2)
 
+        attn_weights = ops.matmul(xq, keys.transpose(2, 3))
+        if self.attention_scale is None:
+            attn_weights = attn_weights / torch.sqrt(self.head_dim)
+        else:
+            attn_weights = attn_weights * self.attention_scale
+
         # Flash attention.
-        if not self.use_grok:
-            attn_weights = ops.matmul(xq, keys.transpose(2, 3)) / math.sqrt(
-                self.head_dim
-            )
-        elif self.use_grok:
-            attn_weights = ops.matmul(xq, keys.transpose(2, 3))
-            attn_weights = 30.0 * torch.tanh(
-                attn_weights * (0.08838834764831845 / 30.0)
```

Comment:
Should we also hardcode the `attention_scale = 0.08838834764831845` in `grok.py`?

Reply:
So the secret is `0.08838834764831845` is not a random hardcoded value, its actually `1.0 / sqrt(128)` and for `grok` `head_dim == 128`. So that normalization is already handled by the default attention scaling above.